src/core/apiLayer/productSearchOptionsService.ts

```diff
@@ -0,0 +1,112 @@
+import { stringify } from "query-string";
+
+import { pkbUrl } from "@temp/constants";
+
+export enum OptionType {
+  ChannelTitle = "channelTitle",
+  JobTitle = "jobTitle",
+  JobFunction = "jobFunction",
+  Location = "location",
+}
+
+export interface Option {
+  type: OptionType;
+  label: string;
+  extraLabel?: string;
+  value: string | number;
+}
+
+const fetchJobTitle = async (text: string): Promise<Option[]> => {
+  const query = stringify({ text, limit: 5 });
+  const headers = { "Accept-Language": "en" }; // @todo: Get from user
+  const response = await fetch(`${pkbUrl}job-titles/?${query}`, { headers })
+    .then(response => response.json())
+    .catch(() => []);
+
+  if (!Array.isArray(response?.results)) {
+    return [];
+  }
+
+  return response.results.map(i => {
+    return {
+      type: OptionType.JobTitle,
+      label: i.name,
+      value: i.id,
+    };
+  });
+};
+
+const fetchJobFunctions = async (text: string): Promise<Option[]> => {
+  const query = stringify({ text });
+  const headers = { "Accept-Language": "en" }; // @todo: Get from user
+  const response = await fetch(`${pkbUrl}job-functions/?${query}`, { headers })
+    .then(response => response.json())
+    .catch(() => []);
+
+  if (!Array.isArray(response)) {
+    return [];
+  }
+
+  return response.slice(0, 4).map(i => {
+    return {
+      type: OptionType.JobFunction,
+      label: i.name,
+      value: i.id,
+    };
+  });
+};
+
+const extractWithin = (location: any, filter: string[]) => {
+  if (!location) {
+    return [];
+  }
+
+  if (!Array.isArray(location.place_type)) {
+    return [];
+  }
+
+  if (!filter.some(i => location.place_type.includes(i))) {
+    return [];
+  }
+
+  return [location.canonical_name, ...extractWithin(location.within, filter)];
+};
+
+const fetchLocations = async (text: string): Promise<Option[]> => {
+  const query = stringify({ text });
+  const headers = { "Accept-Language": "en" }; // @todo: Get from user
+  const response = await fetch(`${pkbUrl}locations/?${query}`, { headers })
+    .then(response => response.json())
+    .catch(() => []);
+
+  if (!Array.isArray(response)) {
+    return [];
+  }
+
+  return response.slice(0, 4).map(i => {
+    return {
+      type: OptionType.Location,
+      label: i.canonical_name,
+      extraLabel:
+        extractWithin(i.within, ["region", "country"]).join(", ") || null,
```

Comment:
why not use `fully_qualified_place_name` key?

Reply:
Because it is emtpy atm on saleor PKB. I already asked @gabalese about it.

Reply:
This is due to the recent translation work we did for PKB on https://github.com/vonq/pkb/pull/413. It will be fixed eventually, put a `//TODO` in there so we don't forget.

Reply:
don't think i get it. @gabalese will syncing with the latest pkb location tables help?

Reply:
And to be honest, the fully_qualified_place_name als contains the canonical_name, I like it better if it does not

Reply:
Then the it would be

> Rotterdam
> Netherlands

and not be

> Rotterdam
> Rotterdam, Netherlands